AdvancedHero.cpp

```diff
@@ -22,6 +22,7 @@ void AdvancedHero::levelup(int levelcount){
     for(int i=0; i<levelcount; i++){
         maxHp = (int)round((double)maxHp * 1.1);
         dmg = (int)round((double)dmg * 1.1);
+        attackspeed = getAttackSpeed()*0.9;
```

Comment:
Az egyik problémám itt is inkább dokumentációs jellegű: ez itt most úgy néz ki, mintha a szintlépéstől csökkenne a támadás sebessége, ahelyett hogy nőne. Ez az érték inkább attackCooldown akar lenni és nem attackSpeed.

A másik probléma, hogy most hiába változik az attackspeed, ennek semmilyen hatása nincs a csatára, mert ott a kezdeti attackspeed másolatával történik a számítás, így hiába változik az attackspeed tagváltozó. Amíg az konstans volt, addig ez nem volt gond, de most már igen. Amúgy is felesleges volt másolattal dolgozni, ami az objektum saját adattagja, azt onnan kérdezzük le mindig, ne egy másolatból.

Reply:
@oliverosz Tehát ha jól értem, akkor az egy jó megoldás lehet, hogy csinálok az ősosztályban egy SetAttackspeed függvényt, és azzal állítom az attackspeedet? 

Értem a dokumentációs problémát, viszont ugyanazt érem el, ha attackcooldown helyett attackspeedet állítanék. Csak az egyikből elveszek 10%-ot, a másiknál meg a 90%-át állítom át minden szintlépésnél. Matematikailag ugyanaz mindkettő. Az egyszerűség kedvéért maradtam ennél a megoldásnál, de a dokumentálás során ezt is megfogom jegyezni.

Reply:
> @oliverosz Tehát ha jól értem, akkor az egy jó megoldás lehet, hogy csinálok az ősosztályban egy SetAttackspeed függvényt, és azzal állítom az attackspeedet?

Nem az a lényeg. Az attackspeedet most is változtatod, nem kell ehhez setter függvény, de nem használod harc közben. A harc lebonyolításának mindig az éppen aktuális attackspeedet kellene figyelembe vennie.

> Értem a dokumentációs problémát, viszont ugyanazt érem el, ha attackcooldown helyett attackspeedet állítanék. Csak az egyikből elveszek 10%-ot, a másiknál meg a 90%-át állítom át minden szintlépésnél. Matematikailag ugyanaz mindkettő.

Matematikailag mindkettő jó lenne, csak az most így kissé félrevezető, hogy a speedet csökkented szintlépésnél, nem növeled. Én átnevezném cooldownra, és akkor megmaradhat a logika, de értelmes neve lesz.